pkg/features/embedded/cloud-config-essentials/system/oem/08_boot_assessment.yaml

```diff
@@ -84,41 +76,36 @@ stages:
             if [ "${enable_boot_assessment}" = "yes" -o "${enable_boot_assessment_always}" = "yes" ]; then
               if [ -z "${selected_entry}" ]; then
                 if [ "${boot_assessment_tentative}" = "yes" ]; then
-                  set default="fallback"
+                  set default="1"
                   set extra_passive_cmdline="upgrade_failure"
                 else
                   set boot_assessment_tentative="yes"
                   save_env -f "${boot_assessment_file}" boot_assessment_tentative
                 fi
               fi
             fi
-    - &efiumount
-      name: "umount EFI"
-      commands:
-      - |
-          umount /tmp/mnt/EFI
 
     # Here we do enable boot assessment for the next bootup.
     # Similarly, we could trigger boot assessment in other cases
     after-upgrade:
-    - <<: *efimount
-    - name: "Set upgrade sentinel"
+    - name: "Set upgrade sentinel on active"
+      if: '[ ! -f "/run/elemental/recovery_mode" ]' 
       commands:
-      - |
-          grub2-editenv /tmp/mnt/EFI/boot_assessment set enable_boot_assessment=yes
+      - grub2-editenv /run/elemental/efi/boot_assessment set enable_boot_assessment=yes
```

Comment:
Isn't `/run/elemental/efi` going to be mounted read-only by default?

Reply:
No, during install we mount everything rw, in ugrades we remount it with RW (state, recovery and EFI) and in reset we do a very similar approach to install (we umount everything but recovery and then mount again with RW later on)

Reply:
To be honest I am not very happy with the boot assessment logic shipped as a custom grub setup, probably we could consider simplifying it by not making it optional and simply including it as part of the default grub configuration.

As soon as we start implementing other snapshotter we will have to revisit grub setup again. Good thing is that nowadays it is way simpler than what it used to be.